src/pageEditor/slices/editorSlice.ts

```
@@ -254,10 +254,10 @@ export const editorSlice = createSlice({
         state.activeElement = null;
       }
 
-      state.elements.splice(
-        state.elements.findIndex((x) => x.uuid === uuid),
-        1
-      );
+      const index = state.elements.findIndex((x) => x.uuid === uuid);
+      if (index > -1) {
```

Under what condition would we expect the UUID to not be found? Should we throw an error in that case?

---

When removing a recipe, we iterate through all the installed extension IDs and call the function returned from the `useRemoveExtension()` hook, which dispatches this redux action, and some of the extensions in the recipe may not have been selected by the user in the side bar, so they are not moved into `elements` yet. I suppose I could also change the call site to select `elements` and then check if the `id` exists before calling this though, if you think that makes more sense.

---

I'll just add a comment here for now